lib/rules/no-magic-numbers.js

```diff
@@ -26,10 +26,73 @@ function normalizeIgnoreValue(x) {
 	return x;
 }
 
+/**
+ * Checks if the node parent is a TypeScript enum member
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if the node parent is a TypeScript enum member
+ */
+function isParentTSEnumDeclaration(node) {
+	return node.parent.type === "TSEnumMember";
+}
+
+/**
+ * Checks if the node is a valid TypeScript numeric literal type.
+ * @param {ASTNode} node The node to be validated
+ * @returns {boolean} True if the node is a TypeScript numeric literal type
+ */
+function isTSNumericLiteralType(node) {
+	let ancestor = node.parent;
+
+	// Go up while we're part of a type union
+	while (ancestor.parent && ancestor.parent.type === "TSUnionType") {
```

Comment:
```suggestion
	while (ancestor.parent.type === "TSUnionType") {
```

It doesn't seem necessary to check if `ancestor.parent` exists here.